app/modules/code_provider/github/github_service.py

```
@@ -18,21 +17,73 @@
+        except Exception as e:
+            logger.warning("Async Redis cache unavailable: %s", e)
+            return None
+
+
+async def close_github_async_redis_cache() -> None:
+    """Close the global async Redis cache. Call from app shutdown to avoid connection leaks."""
+    global _async_redis_cache
+    if _async_redis_cache is not None:
+        try:
+            await _async_redis_cache.aclose()
+        except Exception as e:
```

Ruff (0.15.2)

[warning] Do not catch blind exception: `Exception`

(BLE001)